aztec/src/main/kotlin/org/wordpress/aztec/watchers/event/sequence/ObservationQueue.kt

```diff
@@ -0,0 +1,65 @@
+package org.wordpress.aztec.watchers.event.sequence
+
+import android.os.Build
+import org.wordpress.aztec.watchers.event.IEventInjector
+import org.wordpress.aztec.watchers.event.buckets.API26Bucket
+import org.wordpress.aztec.watchers.event.text.TextWatcherEvent
+import org.wordpress.aztec.watchers.event.buckets.Bucket
+
+class ObservationQueue(injector: IEventInjector) : EventSequence<TextWatcherEvent>() {
+    val buckets = ArrayList<Bucket>()
+    val injector = injector
+
+    init {
+        init()
+    }
+
+    override fun add(element: TextWatcherEvent): Boolean {
+        synchronized(this@ObservationQueue) {
```

Comment:
Same as below, do you think  adding a check on the size of the `buckets` here, and return immediately when = 0, would improve performances on those devices that don't need additional checks?

I know, we will probably add other Buckets soon, to maybe this is not a problem.

Reply:
ah - I see what you mean here! it's probably best to add the check here, and this makes it not necessary to add it in the `processQueue` method, as `add()` is executed before and is the only caller of `processQueue`.
Added in 9117a31